db/models/applications.js

```diff
@@ -96,6 +107,11 @@ class PendingApplication extends Model {
             }
         });
     }
+    static countCompletedApplications(applications) {
```

Comment:
I think this function should live in the application dashboard code as it's not really part of the model (eg. it's just operating on an array of application objects), and then it's in the same place as all the other statistical functions/lookups for the dashboard.

Reply:
:) It was @davidrapson 's suggestion. David - you want to take this one ?

Reply:
I do agree though, that it need not be part of the model as its essentially just a 'count' and can be directly used like `uniqBy` or `getAppsToday` methods on the previous lines without the need for a model/db call.

Reply:
I do think that _something_ like this should be on the model, i.e. the notion of asking for some detail of the application status makes sense on the model to avoid leaking implementation details. Maybe an instance method like:

```js
isComplete() {
   return this.currentProgressState === 'COMPLETE'
}
```

Might be preferable so that we can then write something like

```js
const completedCount = applications.filter(_ => _.isComplete()).length;
```

Having a helper method helps us avoid needing to expose the implementation details of what ENUM value is used.

Reply:
Yeah that makes sense, but keep the actual filtering/counting code with all the rest in applications.js!